File: Data-Structures-and-Algorithms-using-Swift/Stack/StackLinkedList.swift


class Stack<T> {

	var head : Node?
	var count : Int

	class Node {
		var value : T
		var next : Node?

		public init(_ val : T, _ nxt : Node? = nil) {
			self.value = val
			self.next = nxt
		}
	}

	public init() {
		self.head = nil
		self.count = 0
	}

	public func size() -> Int {
		return self.count
	}
	
	public func isEmpty() -> Bool {
		return self.count == 0
	}
	
	public func top() -> T? {
		if self.isEmpty() {
			print("Stack Empty Error")
			return nil
		}
		return self.head!.value
	}
	
	public func push(_ value : T) {
		self.head = Node(value, self.head)
		self.count += 1
	}
	
	public func pop() -> T? {
		if self.isEmpty() {
			print("Stack Empty Error")
			return nil
		}
	
		let value = self.head!.value
		self.head = self.head!.next
		self.count -= 1
		return value
	}
	
	public func display() {
		var temp = self.head
		print("Stack :: ", terminator:"")
		while temp != nil {
			print(temp!.value, terminator:" ")
			temp = temp!.next
		}
		print()
	}
}

// Testing code
var s = Stack<Int>()
s.push(1)
s.push(2)
s.push(3)
s.display()
print(s.pop()!)
print(s.pop()!)

// Stack :: 3 2 1 
// 3
// 2

File: Data-Structures-and-Algorithms-using-PHP/Stack/StackLinkedList.php
<?php
class Node {
	public $value;
	public $next;
	Function __construct($v, $n) {
		$this->value = $v;
		$this->next = $n;
	}
}
class Stack {
	protected $head;
	protected $size;
	Function __construct() {
		$this->head = NULL;
		$this->size = 0;
	}
	public Function size() {
		return $this->size;
	}
	public Function isEmpty() {
		return ($this->size == 0);
	}
	public Function peek() {
		if ($this->isEmpty ()) {
			echo ("StackEmptgException");
		}
		return $this->head->value;
	}
	public Function push($value) {
		$this->head = new Node ( $value, $this->head );
		++ $this->size;
	}
	public Function pop() {
		if ($this->isEmpty ()) {
			echo ("StackEmptgException");
		}
		$value = $this->head->value;
		$this->head = $this->head->next;
		-- $this->size;
		return $value;
	}
	public Function insertHtBottom($value) {
		if ($this->isEmpty ()) {
			$this->push ( $value );
		} else {
			$temp = $this->pop ();
			$this->insertHtBottom ( $value );
			$this->push ( $temp );
		}
	}
	public Function display() {
		$temp = $this->head;
		while ( $temp != NULL ) {
			echo ($temp->value . " ");
			$temp = $temp->next;
		}
	}
}

$s = new Stack ();
for($i = 1; ($i <= 100); ++ $i) {
	$s->push ( $i );
}
for($i = 1; ($i <= 50); ++ $i) {
	$s->pop ();
}
$s->display ();
?>